### PR TITLE
Fix kubectl get -f <file> -o <nondefault printer> so it prints all items in the file

### LINKED ISSUE #38907: kubectl get broken for list
kubectl get -o yaml|json is broken for list.

How to reproduce:
Create multiple resource from list: `kubectl create -f https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/testdata/multi-resource-list.json`.

Get the resources back: `kubectl get -o yaml -f https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/testdata/multi-resource-list.json`

Output:
```yaml
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: 2016-12-16T22:09:53Z
  labels:
    app: mock
  name: mock
  namespace: default
  resourceVersion: "387943"
  selfLink: /api/v1/namespaces/default/services/mock
  uid: 5edec4bb-c3dc-11e6-aa89-42010a800002
spec:
  clusterIP: 10.0.160.121
  ports:
  - port: 99
    protocol: TCP
    targetPort: 9949
  selector:
    app: mock
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```

Only the first resource get printed out.

@kubernetes/kubectl @kubernetes/sig-cli-misc

### PR DESCRIPTION
**Pull request by ncdc**
**What this PR does / why we need it**: Fix kubectl get -f <file> -o <nondefault printer> so it prints all the objects in the file, instead of just the first one. Also add a test for this feature.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #38907 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

cc @AdoHe @deads2k @liggitt @fabianofranz @kubernetes/kubectl @kubernetes/sig-cli-misc

### CODE REVIEW

**Comment by ncdc** [comment]
Setting priority to P1 to match issue

File: pkg/kubectl/cmd/get.go [hunk 1]
```diff
@@ -324,9 +324,7 @@ func RunGet(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args [
 		}
 
 		var obj runtime.Object
-		if singular {
-			obj = infos[0].Object
-		} else {
+		if !singular || len(infos) > 1 {
```
Comment (liggitt) [inline pkg/kubectl/cmd/get.go]: I'm more concerned why singular is true in this case

Comment (ncdc) [inline pkg/kubectl/cmd/get.go]: https://github.com/kubernetes/kubernetes/blob/3484c6f67637792a3a91a5c7ff04cb5b28e56c23/pkg/kubectl/resource/builder.go#L716:

```go
singular := !b.dir && !b.stream && len(b.paths) == 1
```

**Comment by smarterclayton** [comment]
Singular means "give me a single item instead of a list if you can".  "-f
<list>" falls under "cannot", while "-f <item>" falls under "can".

**Comment by jbeda** [comment]
This is very confusing.  The definition of Singular isn't defined.  It is really "SingleItemImplied" where the way things were specified that the user might be thinking of a single object.  If, instead, there are multiple returned users should deal with it.

I'd update the documentation and/or rename `IntoSingular` to make this clearer.

Based on reading, it looks like the change is good but confusing.  Not sure if that is what @smarterclayton was suggesting or not.

**Comment by smarterclayton** [comment]
SingleItemImplied is accurate.  The user asked for one thing, so should get
an object that corresponds to what they asked for (if you ask for a pod,
you get a pod).  The flipside is if you ask for anything that looks like a
list (kubectl get pods, -f <directory>, -f A -f B), you should *always* get
a list.

If you ask for a single item that happens to be a list, you should get a
list.

**Comment by AdoHe** [comment]
We did get a list, and we will flatten this list, handle the item individually, so if `singular` is true, we just print the first object.

**Comment by ncdc** [comment]
Are we saying we need to rename IntoSingular as part of this PR?

**Comment by jbeda** [comment]
I *think* the behavior we want is this:
1. If the user gives us something that ~looks like a single object and actually *is* a single object then print it as a single object.
2. If a user give us something that ~looks like a single object but is actually a list, then print a list.
3. If a user gives us something that looks like a list (recursive directory, etc) then *always* return a list.

The confusion in the code is that `singular` is meant to talk about what the input seems to imply about what the user wants while it is being interpreted in the code as to the number of items returned.

As this is probably not the last time folks will be confused here it might be better to clear up the interface.

**Comment by AdoHe** [comment]
```
The confusion in the code is that singular is meant to talk about what the input seems to imply about what the user wants while it is being interpreted in the code as to the number of items returned.
```
Yes, this is just the truth.

**Comment by AdoHe** [comment]
I think I will carry on this one.

**Comment by ncdc** [comment]
@AdoHe are you still planning on taking over? If not, I can come back to it.

**Comment by smarterclayton** [comment]
I'm fine with a rename.

**Comment by ncdc** [comment]
Ok, cool, I'm picking this back up and will do the rename. FYI @AdoHe

**Comment by ncdc** [comment]
@fabianofranz @smarterclayton @AdoHe updated, ptal

File: pkg/generated/bindata.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func (fi bindataFileInfo) Sys() interface{} {
...
...96\xf0\x89\xfa\x8b\x5b\xb2\x67\xbc\x70\xad\x55\x9b\x77\x9f\xe2\x1e\xba\x80\xa9\x77\x1f\xb9\x8a\x34\x52\xf4\x9e\x7d\xd0\xce\x16\x58\x73\x8c\x7c\x15\xa9\x76\xc4\x57\x72\xd1\x18\x0e\x34\x67\x63\x5c\x72\xca\x5d\x05\x34\x09\xb5\x56\x74\xd0\xd6\x81\x4a\x57\x60\x32\x3d\x29\xe9\xd0\xf3\xea\x4f\xd0\x1b\x19\xb9\xc0\x5c\x64\xdb\x94\xe5\x94\xe5\x98\x8b\x42\x6c\x3f\x13\x42\x88\x64\x7a\x42\xa7\x7c\xa9\xc3\x3f\xdc\xee\x8a\xdb\xc6\x3c\x2b\x5e\xec\x90\xd8\x13\x22\x19\xcb\x10\xa9\xf4\xd2\x06\x23\xa3\xf3\x05\x1e\x78\xb6\x4a\x5a\x3c\x68\xbd\x0d\xf8\x72\xd6\xcb\x54\xa5\xb3\x2e\xf0\xba\x5e\x48\x6d\xd2\xca\x2d\x5e\x25\x93\xd1\xe4\xe8\xa6\x95\x2c\x15\xc9\xa1\xb3\x91\x6d\xa4\xf2\xba\xe1\x02\xbb\xce\x9e\x37\x46\x6a\xbb\x8f\xd5\x5c\xfa\xc0\x71\x78\x56\x1e\xd3\xde\x0d\xd7\xe5\xbd\x60\x4f\x47\xb6\x72\x4a\xdb\xba\xc0\xbd\x99\x8e\xc9\x39\xbd\x65\xcb\xbe\x2f\x68\xea\x58\xe9\x88\x59\xba\x9b\x66\x22\x39\xa7\x5e\xd3\xbb\xd5\x84\x0f\x7b\xdf\x02\x7b\xe3\xb1\xb4\x75\x2b\x6b\xa6\x92\xe5\xa2\x1b\x97\x69\xbd\x34\x74\xec\xfc\x22\x14\x68\x9b\x95\x0c\xc3\x7c\x1f\xfb\xd7\xe1\x96\xc5\x47\x43\xcc\x9e\xee\xaf\x3f\x2d\x90\x6d\x72\xab\xfb\x71\x47\x4b\x83\x4b\x19\xee\x6f\x2b\x7f\x05\x00\x00\xff\xff\xaa\xa3\x15\xc3\x6a\x04\x00\x00")
```
Comment (liggitt) [inline pkg/generated/bindata.go]: revert this file?

Comment (ncdc) [inline pkg/generated/bindata.go]: I dunno - it got updated when I ran make.

Comment (ncdc) [inline pkg/generated/bindata.go]: Ok, undone

**Comment by liggitt** [comment]
seems reasonable to me

**Comment by ncdc** [comment]
Rebased, PTAL!

**Comment by ncdc** [comment]
Lowering priority so #39475 can go in first

**Comment by ncdc** [comment]
Re-adding release-note label because this fixes a regression in 1.5

**Pull request merged**